scrapy_po/utils.py

```diff
@@ -26,19 +26,23 @@ def __init__(self, url, request=None):
 
 def is_callback_using_response(callback: Callable):
     """Check whether the request's callback method is going to use response."""
-    spec = inspect.getfullargspec(callback)
-    try:
-        arg_name = spec.args[1]  # first index is self, second is response
-    except IndexError:
+    if getattr(callback, '__scrapy_po_callback', False) is True:
+        # The callback_for function was used to create this callback.
+        return False
+
+    signature = inspect.signature(callback)
+    first_parameter_key = next(iter(signature.parameters))
```

Comment:
I'm not sure I understand the code well, but could signature.parameters be empty?

Reply:
My understanding: no, because Scrapy requires 'response' to be the first argument.